### PR TITLE
`storage` module: add go docs and minor code quality refactors

### PR DESCRIPTION
**Pull request by raulk**
This absorbs https://github.com/filecoin-project/lotus/pull/2340. Refer to the commit log for details.

### CODE REVIEW

File: chain/store/store.go [hunk 1]
```diff
@@ -1235,6 +1243,9 @@ func (cs *ChainStore) ReadMsgMetaCids(mmc cid.Cid) ([]cid.Cid, []cid.Cid, error)
 	return blscids, secpkcids, nil
 }
 
+// GetPath returns returns the sequence of atomic head change operations that
```
Comment (nonsense) [inline chain/store/store.go:1246]:
```suggestion
// GetPath returns the sequence of atomic head change operations that
```

File: storage/wdpost_run.go [hunk 2]
```diff
@@ -31,6 +31,7 @@ import (
 	"github.com/filecoin-project/lotus/chain/types"
 )
 
+// failPost records a failure in the journal.
 func (s *WindowPoStScheduler) failPost(err error, ts *types.TipSet, deadline *dline.Info) {
```
Comment (nonsense) [inline storage/wdpost_run.go:35]: I find this name a bit confusing. `failPost` sounds like we are failing a PoSt or something like that, given that we are in the `WindowPoStScheduler`.

Wouldn't `recordFailure` or `publishFailure` be better, given that this seems to be an auxiliary method to just log and record a tract of what happened?

File: storage/wdpost_run.go [hunk 3]
```diff
@@ -410,6 +441,12 @@ func (s *WindowPoStScheduler) checkNextFaults(ctx context.Context, dlIdx uint64,
 	return faults, sm, nil
 }
 
+// runPost runs a full cycle of the PoSt process:
+//
+//  1. performs recovery declarations for the next deadline.
+//  2. performs fault declarations for the next deadline.
+//  3. computes and submits proofs, batching partitions and making sure they
+//     don't exceed message capacity.
 func (s *WindowPoStScheduler) runPost(ctx context.Context, di dline.Info, ts *types.TipSet) ([]miner.SubmitWindowedPoStParams, error) {
```
Comment (nonsense) [inline storage/wdpost_run.go:450]: Nit: Similar to my comment from above, wouldn't it be better if this func is called `runPoSt` to make it clear that this is proof of space time?

File: storage/wdpost_run.go (hunk 2 above)
Comment (raulk) [reply storage/wdpost_run.go:35]: Makes sense.

File: storage/wdpost_run.go (hunk 3 above)
Comment (raulk) [reply storage/wdpost_run.go:450]: Yep, let's rename it with the right capitalisation.

**Comment by raulk** [comment]
@nonsense comments addressed!

**Review by nonsense — APPROVED** [review]

File: chain/store/store.go [hunk 4]
```diff
@@ -853,6 +853,14 @@ func (cs *ChainStore) NearestCommonAncestor(a, b *types.TipSet) (*types.TipSet,
 	return cs.LoadTipSet(l[len(l)-1].Parents())
 }
 
+// ReorgOps takes two tipsets (which can be at different heights), and walks
+// their corresponding chains backwards one step at a time until we find
+// a common ancestor. It then returns the respective chain segments that fork
+// from the identified ancestor, in reverse order, where the first element of
+// each slice is the supplied tipset, and the last element is the common
+// ancenstor.
```
Comment (dirkmc) [inline chain/store/store.go:861]:
```suggestion
// ancestor.
```

File: storage/miner.go [hunk 5]
```diff
@@ -152,17 +174,37 @@ func (m *Miner) Run(ctx context.Context) error {
...
-		return m.addrSel.AddressFor(ctx, m.api, mi, use, goodFunds, minFunds)
-	}
+		// Create a shim to glue the API required by the sealing component
+		// with the API that Lotus is capable of providing.
+		// The shim translates between "tipset tokens" and tipset keys, and
+		// provides extra methods.
+		adaptedAPI = NewSealingAPIAdapter(m.api)
+
+		// Instantiate a precommit policy.
+		defaultDuration = policy.GetMaxSectorExpirationExtension() - (md.WPoStProvingPeriod * 2)
```
Comment (dirkmc) [inline storage/miner.go:189]: Suggest adding a comment to explain this calculation

File: storage/miner.go [hunk 6]
```diff
@@ -152,17 +174,37 @@ func (m *Miner) Run(ctx context.Context) error {
...
+		provingBoundary = md.PeriodStart % md.WPoStProvingPeriod
+
+		// TODO: Maybe we update this policy after actor upgrades?
+		pcp = sealing.NewBasicPreCommitPolicy(adaptedAPI, defaultDuration, provingBoundary)
+
+		// address selector.
+		as = func(ctx context.Context, mi miner.MinerInfo, use api.AddrUse, goodFunds, minFunds abi.TokenAmount) (address.Address, abi.TokenAmount, error) {
```
Comment (dirkmc) [inline storage/miner.go:196]: Suggest adding a comment explaining what the address selector does

File: storage/wdpost_run.go [hunk 7]
```diff
@@ -833,6 +879,8 @@ func (s *WindowPoStScheduler) setSender(ctx context.Context, msg *types.Message,
 		minGasFeeMsg.GasFeeCap = msg.GasFeeCap
 	}
 
+	// goodFunds = funds needed for optimal inclusion probability.
```
Comment (dirkmc) [inline storage/wdpost_run.go:882]: 👍 👍

**Review by dirkmc — APPROVED** [review]
A couple of nits but in general looks good 👍

**Pull request merged**